neuraxle/steps/data.py

```diff
@@ -259,7 +260,7 @@ def _transform_data_container(self, data_container: DataContainer, context: Exec
         """
         return self._concatenate_sub_data_containers(data_container)
 
-    def _concatenate_sub_data_containers(self, data_container: DataContainer):
+    def _concatenate_sub_data_containers(self, data_container: DataContainer) -> DataContainer:
```

Comment:
behavior to be reviewed

Reply:
We should phase out these sub_data_container, or at least make it a special DataContainer type.